boot-speed/clouds/measure-cloud.py

```diff
@@ -200,43 +215,65 @@ def gen_datadirname(metadata):
     """ Generate a standardized measurement directory (and tarball) name """
     yyyymmdd = metadata['date']
     cloud = metadata['instance']['cloud']
+    inst_type = metadata['instance']['instance_type']
     release = metadata['instance']['release']
 
-    datadir = Path(cloud + "-" + release + "_" + yyyymmdd)
+    datadir = cloud + "-" + inst_type + "-" + release + "_" + yyyymmdd
     return datadir
 
 
-def main(cloud, release, instances, reboots):
-    if cloud not in known_clouds:
+def main():
+    args = parse_args()
+
+    if args.cloud not in known_clouds:
         print('Unknown cloud provider:', cloud)
         sys.exit(1)
 
+    if args.cloud == 'ec2':
+        instspec = EC2Instspec(args.release)
+        if args.inst_type:
+            instspec.inst_type = args.inst_type
+        if args.region:
+            instspec.region = args.region
+        if args.ec2_subnetid:
+            instspec.subnetid = args.ec2_subnetid
+        if args.ec2_sgid:
+            instspec.sgid = [args.ec2_sgid]
+    else:
+        raise NotImplementedError
+
     tmp_datadir = Path("data")
     tmp_datadir.mkdir()
 
-    if cloud == 'ec2':
-        metadata = measure_ec2(release, tmp_datadir)
-    else:
-        raise NotImplementedError
+    metadata = instspec.measure(tmp_datadir, args.instances, args.reboots)
 
     # str() needed for compatibility with Python <= 3.5
     with open(str(Path(tmp_datadir, "metadata.json")), 'w') as mdfile:
         json.dump(metadata, mdfile)
 
     datadir = gen_datadirname(metadata)
-    with tarfile.open(str(datadir.with_suffix(".tar.gz")), "w:gz") as tar:
+    with tarfile.open((datadir + ".tar.gz"), "w:gz") as tar:
```

Comment:
`datadir` here is presumably not a directory (else we're creating a very strangely named tarball!); could we make the name a little more descriptive?